### PR TITLE
update health check required versions for 3.9

### PR DESCRIPTION
**Pull request by sosiouxme**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1534922

A little refactoring in the first commits to reuse code and prevent having to update this for every single release. Then, updating the actual versions for 3.9.

### CODE REVIEW

File: roles/openshift_health_checker/openshift_checks/__init__.py [hunk 1]
```diff
@@ -309,28 +310,20 @@ def normalize(name_list):
             name_list = name_list.split(',')
         return [name.strip() for name in name_list if name.strip()]
 
-    @staticmethod
-    def get_major_minor_version(openshift_image_tag):
+    def get_major_minor_version(self, openshift_image_tag=None):
         """Parse and return the deployed version of OpenShift as a tuple."""
-        if openshift_image_tag and openshift_image_tag[0] == 'v':
-            openshift_image_tag = openshift_image_tag[1:]
```
Comment (Miciah) [inline roles/openshift_health_checker/openshift_checks/__init__.py]: Did some previous change obviate need to strip any leading 'v'?

File: roles/openshift_health_checker/openshift_checks/__init__.py [hunk 2]
```diff
@@ -325,6 +325,24 @@ def get_major_minor_version(self, openshift_image_tag=None):
...
 
+    def get_required_version(self, name, version_map):
+        """Return the correct required version(s) for the current (or nearest) OpenShift version."""
+        openshift_version = self.get_major_minor_version()
+
+        earliest = min(version_map)
+        latest = max(version_map)
+        if openshift_version < earliest:
+            return version_map[earliest]
+        if openshift_version > latest:
```
Comment (Miciah) [inline roles/openshift_health_checker/openshift_checks/__init__.py]: You could change `<` and `>` to `<=` and `>=`, respectively, but this is fine as is.

File: roles/openshift_health_checker/openshift_checks/__init__.py (hunk 1 above)
Comment (sosiouxme) [reply roles/openshift_health_checker/openshift_checks/__init__.py]: Parsing with a regex instead of splitting on `.`

Comment (Miciah) [reply roles/openshift_health_checker/openshift_checks/__init__.py]: Ah, that would do it.

**Comment by sosiouxme** [comment]
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_openshift-ansible/6757/test_pull_request_openshift_ansible_logging/4758/ is https://github.com/openshift/openshift-ansible/issues/6758

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_openshift-ansible/6757/test_pull_request_openshift_ansible_extended_conformance_install/4011/ looks like https://github.com/openshift/origin/issues/16994

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_openshift-ansible/6757/test_pull_request_openshift_ansible_extended_conformance_install_crio/1126/ is... I really have no idea, but it's unrelated.

/retest
just for the heck of it.

**Comment by sosiouxme** [comment]
enough with the flakes.

**Pull request merged**